src/popup/router/components/TransactionItem.vue

```diff
@@ -50,19 +87,82 @@ export default {
     transaction: { type: Object, required: true },
   },
   computed: {
+    ...mapState('fungibleTokens', ['availableTokens']),
     ...mapGetters([
       'getTxAmountTotal',
       'getTxSymbol',
       'getTxType',
       'getTxDirection',
-      'isTxAex9',
+      'account',
+      'getDexContracts',
+      'getAmountFiat',
     ]),
     txType() {
       return this.getTxType(this.transaction);
     },
-    showType() {
-      return !this.isTxAex9(this.transaction) && this.txType
-        && this.txType !== SCHEMA.TX_TYPE.spend;
+    tokens() {
+      const resolver = TransactionResolver[camelCase(this.transaction?.tx?.function)];
+      if (resolver && !this.isAllowanceTransaction) {
+        return resolver(this.transaction, this.availableTokens).tokens;
+      }
+      return [{
+        ...this.transaction.tx,
+        amount: this.isAllowanceTransaction
+          ? convertToken(this.transaction.tx.fee, -MAGNITUDE)
+          : this.getTxAmountTotal(this.transaction),
+        symbol: this.isAllowanceTransaction ? 'AE' : this.getTxSymbol(this.transaction),
+        isReceived: this.getTxDirection(this.transaction) === 'received',
+        isAe: this.isAllowanceTransaction || this.getTxSymbol(this.transaction) === 'AE',
+      }];
+    },
+    labels() {
```

Comment:
Maybe we could polish this function a bit? I'd suggest putting more complicated ifs in computed. For example 
```suggestion
  isDex() {
  return transactionResolver[camelCase(this.transaction.tx.function)]
        && (this.getDexContracts.wae.includes(this.transaction.tx.contractId)
        || this.getDexContracts.router.includes(this.transaction.tx.contractId))
}, 
    labels() {
    ...
     if (this.isDex) {
     ['DEX', FUNCTION_TYPE_DEX.pool.includes(this.transaction.tx.function) ? 'POOL' : 'SWAP']
     }
    ...
```